slack-java-client/src/main/java/com/hubspot/slack/client/SlackWebClient.java

```diff
@@ -395,7 +399,7 @@ protected Long getInitialOffset() {
   }
 
   private CompletableFuture<Optional<SlackChannel>> findChannelByName(String name, ChannelsFilter channelsFilter) {
-    return searchNextPage(name, listChannels(channelsFilter).iterator());
+    return searchNextPage(name, listChannels(ChannelsListParams.builder().from(channelsFilter).build()).iterator());
```

Comment:
nit: when using a builder, split over multiple lines using indentation to improve the legibility of the code. So this should become:

```java
return searchNextPage(
  name, 
  listChannels(
    ChannelsListParams.builder()
      .from(channelsFilter)
      .build()
    ).iterator()
  );
```